### PR TITLE
Add Dockerfile for neumann

### PR DESCRIPTION
**Pull request by imstar15**
1. Add .ignorefile to speedup building image.
2. Add Dockerfile for neumann 

Run command to build neumann image.
```
docker build -t oaknetwork/neumann:1.2.0 -f ./docker/neumann/Dockerfile .
```

### CODE REVIEW

**Review by irsal — APPROVED** [review]
👍

File: docker/testnet/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM ubuntu:20.04
```
Comment (irsal) [inline docker/testnet/Dockerfile:1]: we don't really need testnet anymore, but this structure is still useful for when we have multiple chains (turing, oak, etc)

File: docker/neumann/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM ubuntu:20.04
+WORKDIR /app
+ADD ./target/release/neumann-collator .
+
+ENTRYPOINT ["./neumann-collator", "-d", "data/node"]
```
Comment (irsal) [inline docker/neumann/Dockerfile:5]:
```
./target/release/neumann-collator --collator --alice --force-authoring --tmp --port 40335 --ws-port 9946 -- --execution wasm --chain resources/rococo-local.json --port 30335
```

See README: https://github.com/OAK-Foundation/OAK-blockchain#launch-the-parachain

**Review by irsal — CHANGES_REQUESTED** [review]
i think there's an issue w/ the way we're building the docker image

**Review by chrisli30 — COMMENTED** [review]
The problem of "Version check failed" has not been resolved. It is related to the protocol definition of OnFinality.

I assume the PR is not done. If yes, let’s add a [WIP] to the title, so reviewers know it’s a Work-in-progress.

**Review by irsal — APPROVED** [review]
thanks for removing the old code

File: docker/neumann/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+LABEL maintainer "contact@oak.tech"
+LABEL description="Binary for Neumann Collator"
+
+# 1. Install curl to execute RPC call
+# 2. Add user
+# 3. Create data directory
+# 4. Link data directory
+# 5. Delete binaries except curl
+RUN apt-get update && apt-get install -y curl && \
```
Comment (imstar15) [inline docker/neumann/Dockerfile:15]: Install curl to request rotateKeys via RPC in docker

File: docker/neumann/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  useradd -m -u 1000 -U -s /bin/sh -d /$USERNAME $USERNAME && \
+  mkdir -p /$USERNAME/.local/share && \
+  mkdir /data && \
+  chown -R $USERNAME:$USERNAME /data && \
+  ln -s /data /$USERNAME/.local/share/$BINARY && \
+  find /usr/bin/* /usr/sbin/* | grep -v curl | xargs rm
```
Comment (imstar15) [inline docker/neumann/Dockerfile:21]: Delete binarys except curl

File: docker/neumann/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+COPY --chown=$USERNAME ./target/$PROFILE/$BINARY /$USERNAME/$BINARY
+COPY --chown=$USERNAME ./resources /$USERNAME/resources
+
+# Open network port
+# 30333 for parachain p2p
+# 30334 for relaychain p2p
+# 9933 for RPC call
+# 9944 for Websocket
+# 9615 for Prometheus (metrics)
+EXPOSE 30333 30334 9933 9944 9615
```
Comment (imstar15) [inline docker/neumann/Dockerfile:35]: 默认暴露了这些端口，如果一台机器用这个image运行多个进程，端口会冲突，需要用-p参数去重新映射。

File: docker/neumann/Dockerfile (hunk 4 above)
Comment (imstar15) [reply docker/neumann/Dockerfile:21]: But this is safer. The mirror image of Moonbeam and Polkadot is the same.

**Pull request merged**